internal/k8s/entity_sort.go

```diff
@@ -0,0 +1,53 @@
+package k8s
+
+// An attempt to order things to help k8s, e.g.
+// a Service should come before things that refer to it.
+// Namespace should be first.
+// In some cases order just specified to provide determinism.
+// Borrowed from Kustomize: https://github.com/kubernetes-sigs/kustomize/blob/b878cd050d5ffe5d485b34b496c915fa96405db9/pkg/gvk/gvk.go#L82
```

Comment:
have we considered linking in the Kustomize code directly?

otherwise, we should probably move this to its own file and add appropriate copyright headers, as in 
https://github.com/windmilleng/tilt/blob/d35c73c006f21488b241ccf58e7c3cc529cd0ec8/internal/kustomize/dependencies.go

Reply:
The Kustomize code doesn't have everything we need, alas -- notably it's missing `PersistentVolume` which is the reason I'm putting in this PR in the first place. i'll add the copyright headers though.